### PR TITLE
[Chore] Component restructure and CI improvement

### PR DESCRIPTION
**Pull request by hanam1ni**
## What happened 👀

This is refactoring PR, there are no changes in the feature
- Extract the survey answer to its own component.
- Make the pagination clickable.
- Update CI: 
  - Cancel the previous workflow.
  - Upload screenshots when UI testings are failing.

## Proof Of Work 📹

- Survey work properly

https://user-images.githubusercontent.com/14077479/177765835-14fabce5-2dd8-4193-89fd-889447e84b70.mov

- When test is failing, the artifact is uploaded: https://github.com/hanam1ni/react-surveyor/actions/runs/2630985181#artifacts

- The previous running workflow is cancel if the new changes are pushed: https://github.com/hanam1ni/react-surveyor/actions/runs/2631152204

### CODE REVIEW

**Comment by malparty** [comment]
Is **[this Action](https://github.com/hanam1ni/react-surveyor/actions/runs/2630985181#artifacts)** a proof of work for the UI Artefacts? If yes, can you update the PR description with this link? Thanks! :)

File: cypress/integration/Home/index.spec.ts [hunk 1]
```diff
@@ -1,13 +1,9 @@
...
-        .trigger('pointermove', 'right')
-        .trigger('pointermove', 'left')
-        .trigger('pointerup');
-      cy.wait(150);
-    }
+  const goToSurveyItem = (index: number) => {
+    cy.get('[data-testid="survey-list"]')
+      .get('.swiper-pagination-bullet')
+      .eq(index - 1)
+      .click();
```
Comment (malparty) [inline cypress/integration/Home/index.spec.ts:6]: Should we keep 1 scroll item so that we can cover both cases (click to switch AND scroll to switch)? 🤔

**Review by malparty — APPROVED** [review]
Minor question/suggestion. The rest LGTM! :) 
Nice changes ~

**Comment by hanam1ni** [comment]
@malparty Updated the description with uploaded artifact and cancel workflow link

File: cypress/integration/Home/index.spec.ts (hunk 1 above)
Comment (hanam1ni) [reply cypress/integration/Home/index.spec.ts:6]: Updated in [f3f0cb2](https://github.com/hanam1ni/react-surveyor/pull/81/commits/f3f0cb27f531c1cdc0f2be34eefe48a0a6a2cb7a)

**Review by hoangmirs — APPROVED** [review]

**Pull request merged**